02stl/cpp/particlesMap.cc

```diff
@@ -0,0 +1,59 @@
+#include <stdio.h>
+#include <map>
+#include <string>
+#include <iostream>
+
+/// func
+void keepMax(std::map<std::string,float>& theMap, const std::string key, const float value)
```

Comment:
In my opinion, functions that mutate arguments should be last resorts only.
Better to do something like 

```
void get_max(xxx const & themap, key, value) {
  auto const existing = themap.find(key);
  return existing != themap.end() ? std::max(existing.second, value): value; 
}
```

And assign outside the function?